frostsnapp/native/src/wallet.rs

```diff
@@ -303,4 +305,227 @@ impl _Wallet {
             }
         })
     }
+
+    pub fn send_to(
+        &mut self,
+        key_id: KeyId,
+        to_address: bitcoin::Address,
+        value: u64,
+        feerate: f32,
+    ) -> Result<TransactionSignTask> {
+        use bdk_coin_select::{
+            metrics, Candidate, ChangePolicy, CoinSelector, DrainWeights, FeeRate, Target,
+            TargetFee, TargetOutputs, TR_DUST_RELAY_MIN_VALUE, TR_KEYSPEND_TXIN_WEIGHT,
+        };
+
+        let utxos: Vec<((KeyId, u32), FullTxOut<_>)> = self
+            .graph
+            .graph()
+            .filter_chain_unspents(
+                &self.chain,
+                self.chain.tip().block_id(),
+                self.graph.index.outpoints().iter().cloned(),
+            )
+            .collect();
+
+        let candidates = utxos
+            .iter()
+            .map(|(_path, utxo)| Candidate {
+                input_count: 1,
+                value: utxo.txout.value,
+                weight: TR_KEYSPEND_TXIN_WEIGHT,
+                is_segwit: true,
+            })
+            .collect::<Vec<_>>();
+
+        let target_output = bitcoin::TxOut {
+            script_pubkey: to_address.script_pubkey(),
+            value,
+        };
+        let mut outputs = vec![target_output];
+
+        let target = Target {
+            fee: TargetFee::from_feerate(FeeRate::from_sat_per_vb(feerate)),
+            outputs: TargetOutputs::fund_outputs(
+                outputs
+                    .iter()
+                    .map(|output| (output.weight() as u32, output.value)),
+            ),
+        };
+
+        // we try and guess the usual feerate from the existing transactions in the graph This is
+        // not a great heuristic since it doesn't focus on transactions the user has sent recently.
+        let long_term_feerate_guess = {
+            let feerates = self
+                .graph
+                .graph()
+                .full_txs()
+                .filter_map(|tx| {
+                    Some(
+                        self.graph.graph().calculate_fee(&tx).ok()? as f32
+                            / tx.weight().to_wu() as f32,
+                    )
+                })
+                .collect::<Vec<_>>();
+
+            let mut average = feerates.iter().sum::<f32>() / feerates.len() as f32;
+
+            if !average.is_normal() {
```

Comment:
Awesome to see coin selection and the change policy in action! 
`f32::is_normal` new terminology to me & useful method.